go/interpreter/lfvm/instructions_test.go

```diff
@@ -880,3 +880,147 @@ func TestExpansionCostOverflow(t *testing.T) {
 		}
 	}
 }
+
+func TestSelfDestruct_StaticModeReportsError(t *testing.T) {
+	ctxt := context{
+		params: tosca.Parameters{
+			Static: true,
+		},
+	}
+	opSelfdestruct(&ctxt)
+	if ctxt.status != statusError {
+		t.Errorf("unexpected status after call, wanted ERROR, got %v", ctxt.status)
+	}
+}
+
+func TestSelfDestruct_Refund(t *testing.T) {
+	tests := map[string]struct {
+		destructed bool
+		revision   tosca.Revision
+		refund     tosca.Gas
+	}{
+		"istanbul": {
+			revision: tosca.R07_Istanbul,
+		},
+		"berlin-first-destructed": {
+			destructed: true,
+			revision:   tosca.R09_Berlin,
+			refund:     24_000,
+		},
+		"berlin-not-first-destructed": {
+			revision: tosca.R09_Berlin,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			refund := selfDestructRefund(test.destructed, test.revision)
+			if refund != test.refund {
+				t.Errorf("unexpected refund, wanted %d, got %d", test.refund, refund)
+			}
+		})
+	}
+}
+
+func TestSelfDestruct_BeneficiaryAccessCost(t *testing.T) {
+	tests := map[string]struct {
+		accessStatus tosca.AccessStatus
+		gas          tosca.Gas
+	}{
+		"cold account": {
+			accessStatus: tosca.ColdAccess,
+			gas:          2_600,
+		},
+		"warm account": {
+			accessStatus: tosca.WarmAccess,
+			gas:          0,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			gas := selfDestructedBeneficiaryAccessCost(test.accessStatus)
+			if gas != test.gas {
+				t.Errorf("unexpected gas, wanted %d, got %d", test.gas, gas)
+			}
+		})
+	}
+}
+
+func TestSelfDestruct_NewAccountCost(t *testing.T) {
+
+	tests := map[string]struct {
+		accountExists bool
+		balance       tosca.Value
```

Comment:
```suggestion
		beneficiaryBalance       tosca.Value
```

Reply:
I believe this comment is about `accountExists` since balance is the current account balance. I will rename `accountExists` to `beneficiaryExists` for clarity